leetcode/2751-robot-collisions.java

```diff
@@ -0,0 +1,60 @@
+package leetcode;
+
+import java.util.*;
+class Solution {
+    class Robot{
```

Comment:
The nested class acts as an entity over here for which no object needs to be created. The usage of the Robot class is limited to the solution class and so for this use-case it would not make much sense defining the Robot class outside the Solution class as it would then violate the OOPs principle of encapsulation.